contracts/src/main/kotlin/com/r3/corda/lib/tokens/contracts/states/FungibleToken.kt

```diff
@@ -33,21 +33,21 @@ import net.corda.core.schemas.QueryableState
  * @param T the [TokenType] this [FungibleToken] state is in respect of.
  */
 @BelongsToContract(FungibleTokenContract::class)
-open class FungibleToken<T : TokenType>(
-        override val amount: Amount<IssuedTokenType<T>>,
+open class FungibleToken(
+        override val amount: Amount<IssuedTokenType>,
         override val holder: AbstractParty,
         override val tokenTypeJarHash: SecureHash? = amount.token.tokenType.getAttachmentIdForGenericParam()
```

Comment:
it needs to be a constructor value, otherwise it is "evaluated" when got. This means that within the verify sandbox, it would be operating on whatever classes were provided to the TX. 

That would allow people to bypass the pinning.

Reply:
IE, the value must be persisted to the ledger rather than a computed value at runtime.